specification/azuredatatransfer/resource-manager/Microsoft.AzureDataTransfer/preview/2023-10-11-preview/azuredatatransfer.json

```diff
@@ -0,0 +1,2361 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2023-10-11-preview",
+    "title": "azuredatatransferrp",
+    "description": "Azure Data Transfer service resource provider"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.AzureDataTransfer/validateSchema": {
+      "post": {
+        "description": "Validates a schema for Azure Data Transfer.",
+        "x-ms-examples": {
+          "Performs action request": {
+            "$ref": "examples/postValidateSchema.json"
+          }
+        },
+        "operationId": "AzureDataTransfer_validateSchema",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "schema",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/schema"
+            },
+            "description": "The schema to validate"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Result of the schema validation.",
+            "schema": {
+              "$ref": "#/definitions/validateSchemaResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.AzureDataTransfer/listApprovedSchemas": {
+      "post": {
```

Comment:
>post [](http://example.com/codeflow?start=7&length=4)

Why do you want to use POST verb for this and not GET? Is there some sort of sensitivity in the data returned?

Reply:
We are creating a [provider-level action](https://armwiki.azurewebsites.net/rpaas/contract/resourcetypecategory.html#4-providerlevelaction) through RPaaS (an endpoint at the provider level and not at a resource level) which needs to be a POST action.

Reply:
@johnkoh-msft Is making it a provider-level proxy (singleton?) resource type, instead of a provider-level action, possible? If thats possible wouldn't it be preferable, so that it can be run with read-only permissions?

Reply:
> @johnkoh-msft Is making it a provider-level proxy (singleton?) resource type, instead of a provider-level action, possible? If thats possible wouldn't it be preferable, so that it can be run with read-only permissions?

This is a good point to consider; however, we believe that the call to "list approved schemas" is more of an "Action" than a "Read" since there are some calculations done on the backend to generate this list. An GET on a [proxy resource](https://armwiki.azurewebsites.net/rpaas/pureproxy.html) called "approvedSchemas" could be possible, but conceptually with our RP, "approvedSchemas" is not associated with a resource definition. We believe a provider-level action aligns more with the action being performed.